bundles/org.dataflowanalysis.analysis.pcm/src/org/dataflowanalysis/analysis/pcm/core/user/CallingUserPCMVertex.java

```diff
@@ -0,0 +1,85 @@
+package org.dataflowanalysis.analysis.pcm.core.user;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.dataflowanalysis.analysis.core.CharacteristicValue;
+import org.dataflowanalysis.analysis.core.DataFlowVariable;
+import org.dataflowanalysis.analysis.pcm.core.AbstractPCMVertex;
+import org.dataflowanalysis.analysis.pcm.core.CallReturnBehavior;
+import org.dataflowanalysis.analysis.resource.ResourceProvider;
+import org.dataflowanalysis.pcm.extension.model.confidentiality.ConfidentialityVariableCharacterisation;
+import org.palladiosimulator.pcm.usagemodel.EntryLevelSystemCall;
+
+public class CallingUserPCMVertex extends UserPCMVertex<EntryLevelSystemCall> implements CallReturnBehavior {
+    private final boolean isCalling;
+
+    /**
+     * Creates a new User Action Sequence Element with an underlying Palladio Element and indication whether the SEFF Action
+     * is calling
+     * @param element Underlying Palladio Element
+     * @param isCalling Is true, when another method is called. Otherwise, a called method is returned from
+     */
+    public CallingUserPCMVertex(EntryLevelSystemCall element, List<? extends AbstractPCMVertex<?>> previousElements, boolean isCalling,
+            ResourceProvider resourceProvider) {
+        super(element, previousElements, resourceProvider);
+        this.isCalling = isCalling;
+    }
+
+    @Override
+    public boolean isCalling() {
+        return this.isCalling;
+    }
+
+    @Override
+    public void evaluateDataFlow() {
+        List<DataFlowVariable> incomingDataFlowVariables = super.getIncomingDataFlowVariables();
+        List<CharacteristicValue> nodeCharacteristics = super.getVertexCharacteristics();
+
+        List<ConfidentialityVariableCharacterisation> variableCharacterisations = this.isCalling
```

Comment:
Please at least write a comment what this does :D

Reply:
Extracted into method with documentation and clearer code in a125c45d0b7df0fe7bdda0760759d54780196dc2